### PR TITLE
adding auto-annotate and inventory compilation scripts, plus new annotation files

### PR DESCRIPTION
**Pull request by nbokulich**
@gregcaporaso would you mind reviewing?

Some big changes here — I generated two draft scripts, ``autoannotate-taxonomy.py`` and ``inventory-compiler.py``, which automatically generate the expected-taxonomy.tsv files, database-identifiers.tsv, and inventory.tsv.

Then I did just that — all bacterial communities have new greengens and silva 97% and 99% OTU expected-taxonomy.tsv files and database-identifiers.tsv files. Others coming next.

This is a complete fix for #22 and #46 and a partial fix for #24

Probably need some clean up and ``inventory-compiler.py`` needs some work to automate whenever specific files are added (if it's possible — see notes).

### CODE REVIEW

File: code/autoannotate.py
Comment (gregcaporaso) [inline code/autoannotate.py]: We generally use [click](http://click.pocoo.org/5/) for our command line interfaces. You might want to look into it, as I think it would reduce a lot of the boilerplate code here, and it would make it easier for others to help with maintenance of this. I think we should transition this from `getopt` to click, but it doesn't necessarily have to happen in this PR. If you want to do it in a different PR, could you create an issue for that?

File: code/autoannotate.py
Comment (gregcaporaso) [inline code/autoannotate.py]: As a convention, you should rename all of your functions to begin with an `_` (so this would become `_add_lists`), unless you want them to be public (meaning that you support other Python programs importing these). If functions are public, then you have to think about backward compatibility, not changing interfaces, etc, so you generally want to make as little public as you can get away with. In this case, since your users would access this functionality though the CLI that you're creating, I highly recommend having no public functions. Users can still import functions that begin with `_`, but it's a general rule that they shouldn't (so it's their fault if they do and their workflows then break).

File: code/autoannotate.py
Comment (gregcaporaso) [inline code/autoannotate.py]: A lot of this is what you wouldn't have to write with click.

File: code/autoannotate.py
Comment (gregcaporaso) [inline code/autoannotate.py]: If `source_fp` doesn't exist, the user is going to see a traceback error. You should detect these types of issues and present more useful error messages.

File: code/autoannotate.py
Comment (gregcaporaso) [inline code/autoannotate.py]: This exact matching is really strict - I wonder if you want to use fuzzy matching instead, to help you identify slight differences in spelling or spelling errors. [Fuzzywuzzy](https://github.com/seatgeek/fuzzywuzzy) is one package that does this. This isn't essential, but is a "nice to have" feature that would probably save you some time.

File: code/autoannotate.py
Comment (gregcaporaso) [inline code/autoannotate.py]: This will overwrite the file if it already exists - is that what you want to do? There's not a right or wrong way to handle this, it's just worth thinking about whether you would want to error if the file already exists to avoid losing work if the user doesn't realize they're about to overwrite a file, or if you just want to overwrite. This decision generally depends on how hard it would be to regenerate the overwritten file if the user didn't really want to overwrite it.

File: data/mock-12/README.md
Comment (gregcaporaso) [inline data/mock-12/README.md]: *more fine-scale variation* -> *more closely related taxa*

Comment (gregcaporaso) [reply data/mock-12/README.md]: *The Extreme strains* -> *These strains*, because you haven't mentioned the name *extreme* yet.

**Review by gregcaporaso — CHANGES_REQUESTED** [review]
Hey @nbokulich,
Some high-level comments... 

The auto-annotate functionality looks *extremely* handy, but I think it's beyond the scope of this specific repo to house this utility code. I can see people wanting to use the code, and then you having to support it, and that's not really what we designed this repo to be. 

A few motivations for putting the code in another repo:
 * It will help keep mockrobiota focused as data resource. It wasn't intended to be a software package, and there are different things that you need to do to manage a software package versus a data resource (e.g., releasing on package managers like pypi is important for software, but not for data resources). The only code that is currently in this repo is to check for data integrity, and I think we should stick with that.
 * As of now there isn't any test code for this, so you really need to be careful with it. If you put it in this repo, which we point users at, we're sort of endorsing use of the code and people *will* start using it. We can't let that happen unless the code has been fully unit tested. 

So, I think you should remove the code directory from this PR, and create a new repository with this code, for example `github.com/nbokulich/mockrobiota-utilities`. That doesn't mean that we could never re-visit this decision, or that you shouldn't use it to help with management of mockrobiota, but it would clearly distinguish the stable data resource from the ultility code that you use to help generate the data files. Once you make that transition, I could give more feedback on the code if you'd like (though transitioning from `getopt` to `click` as I mention in my inline comments would be helpful to do first, since that will reduce the amount of code by a lot). 

Same goes for the other utility scripts that you've included in this pull request. After unit testing, supporting fuzzy matching, etc, I could see the auto-annotate script being publishable as an applications note if we wanted to go that route. So I think you should put these in another repo, use them for a while, and then we should discuss whether it's worth trying to make them into a tool that we'd publicly support/publish (in which case it might be time to put them into `mockrobiota`, though that's a decision we'd want to weigh at the time) or whether we just want them to remain utilities that are not support for use by others. 

Other high-level comments:
 - [x] could you rename the otu directories to use dashes instead of underscores (so `99-otus` rather than `99_otus`) for consistency with the other directory names in the repo? It's a pain to mix both as you can never remember/guess which will be used at any given time. (We like dashes because you don't have to hit *shift*... yep, programmers are that lazy...) 
 - [x] remove `data/mock-12/.DS_Store`, and any other `.DS_store` files that are hiding around the repo. If you add `.DS_store` to the `.gitignore` file, git won't try to track them anymore. [Here's an example.](https://github.com/qiime2/qiime2/blob/8acf1e2e6f618d72519f310850083f6b01dcfacc/.gitignore#L67)
 - [x] I think you should rename `data/mock-12/source/ExtremeRefSeqs.fasta` to `data/mock-12/source/expected-sequences.fasta`, for consistency with `expected-taxonomy.tsv`. What do you think of that? Simiarly, `HMP_MOCK.fasta` should become `mock.1.fasta` (and in general, the names should be <sample-id>.fasta). 
 
The additional data sets are fantastic, and it's awesome that we have sequence data for some now! 

One other comment: it's generally a good idea to keep your pull requests more modular. In this case, I would have made the changes to all of the data one PR, the addition of the new data sets another PR, and the addition of the code into a third PR. This makes them easier to review, and in this case would mean less work because we could just not merge the PR that adds the `code` directory, rather than having to revert the addtion in this PR. 

I can review again when you've made these changes. Thanks for the work on this!

**Comment by nbokulich** [comment]
Thanks @gregcaporaso ! I have pushed all requested changes (with the exception of code edits, as these are now moved to a new repo).

**Review by gregcaporaso — CHANGES_REQUESTED** [review]
This all looks good, but I noticed one more thing:

- [ ] `mockrobiota/data/example-1/greengenes/13_8/97/expected-taxonomy.tsv` have `97-otus` as the lowest level directory (ie., not just `97`? 

Once that change is made this is ready for merge.

**Comment by nbokulich** [comment]
Thanks @gregcaporaso ! Have made that fix and pushed the changes.

**Review by gregcaporaso — APPROVED** [review]

**Comment by gregcaporaso** [comment]
Thanks @nbokulich, looks good!

**Pull request merged**
